### ISSUE TITLE
Not able to install pkg

### ISSUE THREAD

**Issue by omkarpai** [yao-pkg/pkg#128]
### What version of pkg are you using?

Not applicable

### What version of Node.js are you using?

v22.12.0

### What operating system are you using?

Ubuntu 22.04

### What CPU architecture are you using?

x86_64

### What Node versions, OSs and CPU architectures are you building for?

Not applicable

### Describe the Bug

Running
```npm install -g @yao-pkg/pkg```

Results in 
```
npm warn tar TAR_BAD_ARCHIVE: Unrecognized archive format
npm error code TAR_BAD_ARCHIVE
npm error TAR_BAD_ARCHIVE: Unrecognized archive format
npm error A complete log of this run can be found in: /home/omkar/.npm/_logs/2024-12-17T10_18_18_571Z-debug-0.log
```

Full log `/home/omkar/.npm/_logs/2024-12-17T10_18_18_571Z-debug-0.log`
[2024-12-17T10_18_18_571Z-debug-0.log](https://github.com/user-attachments/files/18163906/2024-12-17T10_18_18_571Z-debug-0.log)

Error snippets from log:
```
944 http fetch GET 200 https://registry.npmjs.org/tr46/-/tr46-0.0.3.tgz 133ms (cache miss)
945 http fetch GET 200 https://registry.npmjs.org/y18n/-/y18n-5.0.8.tgz 137ms (cache miss)
946 http fetch GET 304 https://registry.npmjs.org/fs-constants/-/fs-constants-1.0.0.tgz 156ms (cache skip)
947 warn tar TAR_BAD_ARCHIVE: Unrecognized archive format
948 silly tar Error: TAR_BAD_ARCHIVE: Unrecognized archive format
948 silly tar     at Unpack.warn (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/warn-mixin.js:21:40)
948 silly tar     at Unpack.warn (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/unpack.js:236:18)
948 silly tar     at Unpack.<anonymous> (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:83:14)
948 silly tar     at Unpack.emit (node:events:536:35)
948 silly tar     at [emit] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:313:12)
948 silly tar     at [maybeEnd] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:468:17)
948 silly tar     at [consumeChunk] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:476:21)
948 silly tar     at Unpack.write (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:427:25)
948 silly tar     at Unpack.end (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:548:14)
948 silly tar     at Pipe.end (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/minipass/dist/commonjs/index.js:116:23) {
948 silly tar   recoverable: false,
948 silly tar   cwd: '/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/@yao-pkg/pkg/node_modules/fs-constants',
948 silly tar   code: 'TAR_BAD_ARCHIVE',
948 silly tar   tarCode: 'TAR_BAD_ARCHIVE'
948 silly tar }
949 http fetch GET 200 https://registry.npmjs.org/ieee754/-/ieee754-1.2.1.tgz 163ms (cache miss)

```

```
1080 http fetch GET 200 https://registry.npmjs.org/resolve/-/resolve-1.22.9.tgz 822ms (cache miss)
1081 verbose stack Error: TAR_BAD_ARCHIVE: Unrecognized archive format
1081 verbose stack     at Unpack.warn (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/warn-mixin.js:21:40)
1081 verbose stack     at Unpack.warn (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/unpack.js:236:18)
1081 verbose stack     at Unpack.<anonymous> (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:83:14)
1081 verbose stack     at Unpack.emit (node:events:536:35)
1081 verbose stack     at [emit] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:313:12)
1081 verbose stack     at [maybeEnd] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:468:17)
1081 verbose stack     at [consumeChunk] (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:476:21)
1081 verbose stack     at Unpack.write (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:427:25)
1081 verbose stack     at Unpack.end (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/tar/lib/parse.js:548:14)
1081 verbose stack     at Pipe.end (/home/omkar/.nvm/versions/node/v22.12.0/lib/node_modules/npm/node_modules/minipass/dist/commonjs/index.js:116:23)
1082 error code TAR_BAD_ARCHIVE
1083 error TAR_BAD_ARCHIVE: Unrecognized archive format
1084 silly unfinished npm timer reify 1734430698794
1085 silly unfinished npm timer reify:unpack 1734430701527
1086 silly unfinished npm timer reifyNode:node_modules/@yao-pkg/pkg/node_modules/fs-constants 1734430701527
1087 verbose cwd /home/omkar
1088 verbose os Linux 6.8.0-49-generic
1089 verbose node v22.12.0
1090 verbose npm  v10.9.0

```

Same behaviour is observed with prebuilt node binaries (https://nodejs.org/en/download/prebuilt-binaries) as well.

Initially this was observed with Node 14.21.3. Suspecting it was a node version issue, I tried using Node 22 but have also encountered the same issue.

### Expected Behavior

Expected pkg to get installed.

### To Reproduce

Have fresh installs of node & npm with nvm or do a cache clean.
```
nvm use 22
npm cache clean --force
npm install -g @yao-pkg/pkg
```

**Comment by robertsLando** [comment]
```
npm install -g @yao-pkg/pkg

added 151 packages in 3s

30 packages are looking for funding
  run `npm fund` for details
npm notice
npm notice New major version of npm available! 10.8.2 -> 11.0.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.0.0
npm notice To update run: npm install -g npm@11.0.0
npm notice
```

It works on my side so 🤷🏼‍♂️

**Comment by omkarpai** [comment]
seems like it was an issue with npm's cdn causing issues in the Mumbai region.
https://status.npmjs.org/
